packages/fxa-content-server/app/scripts/models/account.js

```diff
@@ -263,6 +229,53 @@ const Account = Backbone.Model.extend({
       });
   },
 
+  /**
+   * Create an OAuth code
+   * @param {String} clientId
+   * @param {String} state
+   * @param {Object} [options={}]
+   *   @param {String} [options.access_type=online] if `access_type=offline`, a refresh token
+   *     will be issued when trading the code for an access token.
+   *   @param {String} [options.acr_values] allowed ACR values
+   *   @param {String} [options.keys_jwe] Keys used to encrypt
+   *   @param {String} [options.redirect_uri] registered redirect URI to return to
+   *   @param {String} [options.response_type=code] response type
+   *   @param {String} [options.scope] requested scopes
+   *   @param {String} [options.code_challenge_method] PKCE code challenge method
+   *   @param {String} [options.code_challenge] PKCE code challenge
+   * @returns {Promise} A promise that will be fulfilled with:
+   *   - `redirect` - redirect URI
+   *   - `code` - authorization code
+   *   - `state` - state token
+   */
+  createOAuthCode (clientId, state, options) {
+    const sessionToken = this.get('sessionToken');
+    if (! sessionToken) {
+      return Promise.reject(AuthErrors.toError('INVALID_TOKEN'));
+    }
+
+    return this._fxaClient.createOAuthCode(sessionToken, clientId, state, options);
```

Comment:
Out of curiosity, why is `state` pulled out into a standalone argument here while other oauth params are in an options hash? I don't feel strongly about it, just something I didn't quite follow at first.

Reply:
Because state is always required whereas the others are optional.

Reply:
> Because state is always required whereas the others are optional.

https://github.com/mozilla/fxa/blob/cf0435e4b20012e48f89234c0d99bdca3efef755/packages/fxa-auth-server/lib/oauthdb/create-authorization-code.js#L23

Reply:
The above is from the auth-server, in the oauth-server `state` is optional when `response_type=token`:

https://github.com/mozilla/fxa/blob/cf0435e4b20012e48f89234c0d99bdca3efef755/packages/fxa-auth-server/fxa-oauth-server/lib/routes/authorization.js#L64

It looks like we have a discrepancy in the APIs, is this intentional or an oversight?